app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue

```diff
@@ -42,22 +46,36 @@ export default {
       if (newValue !== oldValue) {
         this.viewer.clear();
         this.viewer.setAttribute('columns', JSON.stringify(newValue))
-        this.initPerspective(this.items);
+        this.checkIfQueryResultIsEmpty(this.items)
       }
     }
   },
   mounted() {
     this.viewer = this.$refs["perspective-viewer"];
-    this.initPerspective(this.items);
+    this.checkIfQueryResultIsEmpty(this.items)
     this.listenerPerspective()
   },
   methods: {
+    // You can run a query that gets an empty result, and this isn't an error. But if the result comes empty Perspective has no data to build the table, so console returns an error. We need to check if the result of the query is equal to the columns
+    checkIfQueryResultIsEmpty(items) {
+      //The result of the query, if returns empty only contains the columns.
+      const data = items.trim()
+      //Only the columns of the query.
+      const arrayColumnsQueryString = this.arrayColumnsQuery.toString()
+
+      if (arrayColumnsQueryString !== data) {
+        this.initPerspective(this.items)
+      } else {
+        this.viewer.clear()
+        // Well, it's a bit tricky, but reset the table with .clear() only responds when trigger an event, if not trigger an event .clear() isn't fired
+        window.dispatchEvent(new Event('resize'))
```

Comment:
I see, but why is required to _resize_? What for?

Reply:
I tried it with some events like scroll, click, doublick, focus and doesn't works, but Perspective reacts/listened to the Resize event

Reply:
Nop, my question is about why do we need an event? Only to make it work `this.viewer.clear()`? So you're saying that clause do nothing until you trigger an event? That's weird, so I asked.